storage/innobase/fil/fil0crypt.cc

```diff
@@ -3144,3 +3145,169 @@ fil_space_verify_crypt_checksum(
 
 	return(encrypted);
 }
+
+redo_log_key* redo_log_keys::load_latest_key() {
+	size_t klen = 0;
+	char *key_type = NULL;
+	byte *rkey = NULL;
+
+	if (my_key_fetch(PERCONA_REDO_KEY_NAME, &key_type, NULL,
+		reinterpret_cast<void**>(&rkey), &klen) || rkey == NULL ||
+		strncmp(key_type, "AES", 4) != 0) {
+		
+		/* There is no key yet, we'll try to generate one */
+		my_free(rkey);
+		return generate_and_store_new_key();
+	}
+
+
+	uint    version = 0;
```

Comment:
Formatting. I asked for specifically clang-format at the first review already, there are simply too many things to fix manually. 5.7 has .clang-format. If the problem is that the rest of fil0crypt.cc is unformatted, surround with "noformat" comments before clang-format invocation as needed

Reply:
There is a clang format, but it doesn't match the code exactly. The clang-format pre-commit hook script results in a 73k line diff file for this patch, and 99% of that isn't my change.

For encryption related parts, it's even worse than usual.

Reply:
Right, because all of fil0crypt is a dumster fire as far as formatting is concerned, and we will have to commit that 73K diff sooner or later.

That's why I suggested noformat as a workaround

Reply:
I'm trying to do that now, but it's not only that. Clang-format would reformat huge parts of most innodb files I touched (e.g. 2k line diff for os0file.h), and for some files (such as the system_key header) would completely rearrange the code.

Reply:
I've reformatted everything redo log encryption related with clang-format. I left a few things (such as system_key) alone. I think clang-format also formats class/struct declarations incorrectly for innodb, but I didn't change that.